src/arithmetic/agg_funcs.c

```diff
@@ -197,10 +214,76 @@ AggCtx* Agg_CountFunc() {
 
 //------------------------------------------------------------------------
 
+typedef struct {
+    double percentile;
+    double *values;
+    size_t count;
+} __agg_percCtx;
+
+// This function is agnostic as to percentile method
+int __agg_percStep(AggCtx *ctx, SIValue *argv, int argc) {
+    __agg_percCtx *ac = Agg_FuncCtx(ctx);
+
+    // The first argument is the percentile value. This only actually needs to be set once
+    // (and is not actually used except in Reduce)
+    if (!SIValue_ToDouble(&argv[0], &ac->percentile)) {
+        return Agg_SetError(ctx,
+                "PERC_DISC Could not convert percentile argument to double");
+    }
+
+    // The values buffer will be resized to handle 1000 more values every time
+    // its capacity is met
+    if (ac->count + 1 % 1000 == 0) {
+        realloc(ac->values, ac->count + 1000);
```

Comment:
1000 was arbitrarily chosen for the length default and increment. A better practice might be to double the buffer size every time `ac->count` is met.